src/app/schemas/robot_type.py

```diff
@@ -3,25 +3,34 @@
 
 
 class RobotTypeBase(BaseModel):
-    name: str = None
-    vendor: Optional[str] = None
+    name: str
+    vendor: str
+    capacity_load_kg: Optional[float] = None
+    range_m: Optional[float] = None
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "name": "Niryo One",
+                "vendor": "Niryo",
+                "capacity_load_kg": 50,
+                "range_m": 3
+            }
+        }
 
 
 class RobotTypeCreate(RobotTypeBase):
-    name: str
-    vendor: str
+    pass
 
 
 class RobotTypeUpdate(RobotTypeBase):
-    name: str
-    vendor: str
+    name: Optional[str] = None
+    vendor: Optional[str] = None
```

Comment:
This means an update with body `{ "name": "NiryoOne" }` will leave `vendor` as is (and don't set it to `None`), correctly?

Reply:
Should not change it, yes. Either way, it was a leftover and not intended. Thanks for the catch! Fix is already squashed in.

Reply:
Should not change it, yes. Either way, it was a leftover and not intended. Thanks for the catch! Fix is already squashed in.